### ISSUE TITLE
A way to detect is tested object is a promise at all

### ISSUE THREAD

**Issue by jrencz** [bvaughn/jasmine-promise-matchers#13]
I'd love to see something that will facilitate testing whether something is a promise at all - not whether it's resolved or rejected.

So far I came up with:

```javascript
expect(promiseOrNot.constructor).toBe($q(function() {}).constructor);
```

(off course after checking if it's an object or not)

@bvaughn do you find a matcher `toBePromise` useful and worth adding to this package?

**Comment by bvaughn** [comment]
@jrencz Yeah, I think this would be a nice addition. Would you be interested in contributing a PR?

**Comment by jrencz** [comment]
@bvaughn I'll try, but I don't know the internals of the project yet. It may take a while

**Comment by bvaughn** [comment]
No worries. The project is actually *very* small. The entire source is in one file, [`source/source.js`](https://github.com/bvaughn/jasmine-promise-matchers/blob/develop/source/source.js). The rest of the project is just setup to test that source with Jasmine 1.3 and 2.x.

I would normally be happy to do this myself but I recently joined a startup and I'm *slammed* lately.

**Comment by bvaughn** [comment]
Seeing this issue again and still thinking it would be nice to have, so I think I'll add it.

Looks like Angular does this type of check internally [like this](https://github.com/angular/angular.js/blob/d653607162488839b440de7b00b3c00d360bca3a/src/ng/q.js#L334):
```javascript
if (value && isFunction(value.then)) return value;
```

**Comment by bvaughn** [comment]
By the way @jrencz, this feature is now available in release 1.1.3 :)
